A92-xds-ext-authz.md

```diff
@@ -0,0 +1,220 @@
+    not present in the bootstrap config, we will reject the config.
+  - [allow_envoy](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/config/common/mutation_rules/v3/mutation_rules.proto#L59):
+    TODO: Do we need this?  Those headers aren't really special to gRPC.
+  - [disallow_system](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/config/common/mutation_rules/v3/mutation_rules.proto#L65):
+    TODO: Should we allow these headers to be modified at all?
+  - [disallow_all](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/config/common/mutation_rules/v3/mutation_rules.proto#L70)
+  - [allow_expression](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/config/common/mutation_rules/v3/mutation_rules.proto#L75):
+    TODO: Figure out how to handle matching a restricted header.  Maybe just
+    don't allow it to match against those?
+  - [disallow_expression](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/config/common/mutation_rules/v3/mutation_rules.proto#L79)
+  - [disallow_is_error](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/config/common/mutation_rules/v3/mutation_rules.proto#L87)
+- [include_peer_certificate](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto#L181)
+- TODO: CACHING!
+
+The following fields will be ignored by gRPC:
+- http_service: It doesn't make sense for gRPC to support non-gRPC
+  mechanisms for contacting the ext_authz server.
+- stat_prefix, charge_cluster_response_stats, emit_filter_state_stats:
+  These do not apply to gRPC.
+- transport_api_version: Not relevant, since gRPC supports only xDS v3.
+- with_request_body: This feature is structured around an HTTP request
+  and doesn't really make sense for gRPC, both because it's unclear how
+  it would work for streaming RPCs and because it would include only the
+  first message on the stream (after gRPC deframing), not the raw HTTP
+  DATA frame content.
+- validate_mutations: We will unconditionally reject invalid mutations.
+- clear_route_cache: We don't currently support recomputing the route.
+  We could consider adding this in the future if we have a use-case for
+  it.
+- encode_raw_headers: We will unconditionally encode headers in raw form.
+- metadata_context_namespaces, typed_metadata_context_namespaces,
+  route_metadata_context_namespaces, route_typed_metadata_context_namespaces,
+  enable_dynamic_metadata_ingestion, filter_metadata: gRPC does not currently
+  support dynamic metadata.
+- bootstrap_metadata_labels_key: We have no current use-case for this.  We
+  could consider adding it in the future if there is a need.
+- filter_enabled, filter_enabled_metadata, deny_at_disable: We don't currently
```

Comment:
Regarding `RuntimeFractionalPercent filter_enabled` - we could use the [`type.v3.FractionalPercent default_value`](https://github.com/envoyproxy/envoy/blob/82bc63199ff429490260e52794bb3095f17bcdae/api/envoy/config/core/v3/base.proto#L648) to configure the filter. That was my plan for RLQS anyway.

Reply:
My thinking was "we don't have the runtime, so we fall back to `default_value`, as expected".

Reply:
Yeah, we do the same thing in the fault injection filter.  I was thinking it was redundant here with the generic mechanism for filter disabling, but I guess it's still useful for enabling the filter based for only a percentage of traffic.

I've added support for both `filter_enabled` and `deny_at_disable`.